### PR TITLE
Support additional keys in projects.

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

File: src/build/project.c [hunk 1]
```diff
@@ -351,6 +355,12 @@ static void load_into_build_target(JSONObject *json, const char *type, BuildTarg
 	const char *cpu = get_valid_string(json, "cpu", type, false);
 	if (cpu) target->cpu = cpu;
 
+	// Force linker
+	target->force_linker = get_valid_bool(json, "forcelinker", type, target->force_linker);
+
+	// Safe mode
+	target->feature.safe_mode = get_valid_bool(json, "fast", type, target->feature.safe_mode);
```
Comment (lerno) [inline src/build/project.c:362]: This looks the opposite of how it works, i.e. if you pick "fast" you get "safe"

Comment (data-man) [reply src/build/project.c:362]:
```c
target->feature.safe_mode = get_valid_bool(json, "fast", type, !target->feature.safe_mode);
```

Comment (data-man) [reply src/build/project.c:362]: Or change it to `safe`?

**Comment by lerno** [comment]
Leave out the "safe", we'll reconsider how that should look.

**Comment by lerno** [comment]
There is still "fast" now. Neither should be in.

File: src/build/project_creation.c [hunk 2]
```diff
@@ -31,9 +31,13 @@ const char* JSON_EXE =
 		"  \"targets\": {\n"
 		"    \"%s\": {\n"
 		"      // executable or library\n"
-		"      \"type\": \"executable\"\n"
+		"      \"type\": \"executable\",\n"
 		"      // additional libraries, sources\n"
 		"      // and overrides of global settings here\n"
+		"       \"opt\": \"O0\",\n"
```
Comment (lerno) [inline src/build/project_creation.c:37]: These should probably be commented out or?

**Comment by lerno** [comment]
I've now updated, so you can add "optsize" "optlevel" "safe" to the project.

**Pull request merged**